### PR TITLE
[Merged by Bors] - Feat: port Data.Countable.Basic

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #1745

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1745
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1745~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Countable/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+#align nonempty_embedding_nat nonempty_embedding_nat
+
+protected theorem Function.Embedding.countable [Countable β] (f : α ↪ β) : Countable α :=
+  f.injective.countable
+#align function.embedding.countable Function.Embedding.countable
+
+end Embedding
+
+/-!
+### Operations on `Type*`s
```
Comment (jcommelin) [inline Mathlib/Data/Countable/Basic.lean:52]:
```suggestion
### Operations on `Type _`s
```

File: Mathlib/Data/Countable/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  exact ((Equiv.sigmaEquivProd ℕ ℕ).injective.comp <| hf.sigma_map hg).countable
+
+end type
+
+section sort
+
+variable {α : Sort u} {β : Sort v} {π : α → Sort w}
+
+/-!
+### Operations on and `Sort*`s
```
Comment (jcommelin) [inline Mathlib/Data/Countable/Basic.lean:84]:
```suggestion
### Operations on `Sort _`s
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**